README.md

```diff
@@ -1,41 +1,138 @@
-# spectral
+# Spectral
 
-> **Warning** This is still a prototype and not ready for general use.
+[![Maintainability](https://api.codeclimate.com/v1/badges/1aa53502913a428f40ac/maintainability)](https://codeclimate.com/github/stoplightio/spectral/maintainability)
+[![Test Coverage](https://api.codeclimate.com/v1/badges/1aa53502913a428f40ac/test_coverage)](https://codeclimate.com/github/stoplightio/spectral/test_coverage)
 
-This is an enhanced version of the [speccy](https://github.com/wework/speccy)
-project. Differences between this project and speccy include:
+## Features
 
-- Lint rules can be applied to _any_ JSON object, not just OAS3 specifications.
+- Allows you to create custom rules to lint _any JSON object_
+- Built-in set of functions to help build custom rules, like for patterns, parameter checks, alphabetical ordering, a specified number of characters, provided keys are present in an object, etc.
+- Existing rulesets for OpenAPI Specification (OAS) 2 _and_ 3 documents
+- Validates JSON with [Ajv](https://github.com/epoberezkin/ajv)
 
-- All dependencies on the [oas-kit](https://github.com/Mermade/oas-kit/)
-  repository have been removed, since rules are no longer OAS-specific.
+## Installation
 
-- The rule structure has been modified slightly to use
-  [JSONPath](http://goessner.net/articles/JsonPath/) `path` parameters instead
-  of the `object` parameters (which were OAS-specific).
+```shell
+npm install @stoplight/spectral
+```
 
-- Rules are more clearly defined (thanks to TypeScript typings) and now require
-  specifying a `type` parameter.
+Supports Node v8.3+
 
-- Some rule types have been enhanced to be a little more flexible. An example of
-  this includes the ability to specify the object to be linted in the `path`
-  parameter itself, instead of relying on rule-specific options to be applied.
+## Usage
 
-- Ported to TypeScript.
+There are three key parts to working with Spectral: **Functions**, **rules**, and **rulesets**. **Rulesets** are made up of **rules**, which are built with **functions**.
+
+Think of a **ruleset** as your own flexible and customizable style guide for your JSON objects.
+
+### Creating a custom rule and ruleset:
+
+Spectral has a built-in set of functions, which you can write your own rules with. This example uses the `RuleFunction.PATTERN` to create a rule that checks that all property values are in snake case.
+
+```javascript
+const { Spectral } = require("@stoplight/spectral");
+
+const rulesets = {
+  rulesets: [
+    {
+      rules: {
+        mySpec: {
+          snake_case: {
+            type: RuleType.STYLE,
+            enabled: true,
+            summary: 'Checks for snake case pattern',
+            function: RuleFunction.PATTERN,
```

Comment:
Suggest add `console.log(results)` at end of example, and then past into a js file and `node example.js`. You'll see a couple of issues :). First being that `RuleFunction` is not imported.

Should do this exercise with all of the examples - end user should be able to copy paste them and have them work when run file with node.